src/test/java/com/microsoft/sqlserver/jdbc/fmtOnly/SelectTest.java

```diff
@@ -0,0 +1,241 @@
+        valuePair.add(Pair.of("SELECT * FROM [dbo].[jdbcDB].[jdbctest]", "[dbo] . [jdbcDB] . [jdbctest]"));
+        valuePair.add(Pair.of("SELECT * FROM [dbo].\"jdbcDB\".\"jdbctest\"", "[dbo] . \"jdbcDB\" . \"jdbctest\""));
+        valuePair.add(Pair.of("SELECT * FROM [jdbc test]", "[jdbc test]"));
+        valuePair.add(Pair.of("SELECT * FROM [jdbc /*test*/]", "[jdbc /*test*/]"));
+        valuePair.add(Pair.of("SELECT * FROM [jdbc //test]", "[jdbc //test]"));
+
+        // with parameters
+        valuePair.add(Pair.of("SELECT c1,c2,c3 FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT (c1,c2,c3) FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT ?,?,? FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT (c1,?,c3) FROM jdbctest", "jdbctest"));
+
+        // with special parameters
+        valuePair.add(Pair.of("SELECT [c1],\"c2\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [c1]]],\"c2\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [c]]1],\"c2\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [c1],\"[c2]\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [\"c1],\"FROM\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [\"c\"1\"],\"c2\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT ['FROM'1)],\"c2\" FROM jdbctest", "jdbctest"));
+        valuePair.add(Pair.of("SELECT [((c)1{}],\"{{c}2)(\" FROM jdbctest", "jdbctest"));
+
+        // with sub queries
+        valuePair.add(Pair.of(
+                "SELECT t.*, a+b AS total_sum FROM (SELECT SUM(col1) as a, SUM(col2) AS b FROM table ORDER BY a ASC) t ORDER BY total_sum DSC",
+                "(SELECT SUM (col1 )as a , SUM (col2 )AS b FROM table ORDER BY a ASC ) t"));
+        valuePair.add(Pair.of("SELECT col1 FROM myTestInts UNION "
+                + "SELECT top 1 (select top 1 CONVERT(char(10), max(col1),121) a FROM myTestInts Order by a) FROM myTestInts Order by col1",
+                "myTestInts UNION SELECT top 1 (select top 1 CONVERT (char (10 ), max (col1 ), 121 )a FROM myTestInts Order by a ) FROM myTestInts"));
+
+        // Multiple Selects
+        valuePair.add(Pair.of("SELECT * FROM table1;SELECT * FROM table2", "table1,table2"));
+        valuePair.add(Pair.of("SELECT * FROM table1;SELECT * FROM table1", "table1"));
+
+        valuePair.forEach(p -> assertEquals(p.getRight(), ParserUtils.getTableName(p.getLeft()).trim()));
+    }
+
+    /*
+     * https://docs.microsoft.com/en-us/sql/t-sql/queries/select-examples-transact-sql?view=sql-server-2017
+     */
+    @Test
+    public void selectExamplesTest() {
+        ArrayList<Pair<String, String>> valuePair = new ArrayList<>();
+        // A. Using SELECT to retrieve rows and columns
+        valuePair.add(Pair.of("SELECT * FROM Production.Product ORDER BY Name ASC;", "Production . Product"));

```

Comment:
This gets very long and looks cluttered and kinda hard to read when the query is long and is also harder to find out which query had failed when there are failures. Do you think it might be better to have a helper function which takes in 2 strings and do the assertEqual right away? Then we know immediately the line that caused the failure.